storage/pruning/pruningStorer.go

```diff
@@ -203,15 +211,18 @@ func initPersistersInEpoch(
 		return createPersisterIfPruningDisabled(args, shardIDStr)
 	}
 
-	if args.NumOfEpochsToKeep < args.NumOfActivePersisters {
+	if args.EpochsData.NumOfEpochsToKeep < args.EpochsData.NumOfActivePersisters {
 		return nil, nil, fmt.Errorf("invalid epochs configuration")
 	}
 
-	oldestEpochActive, oldestEpochKeep := computeOldestEpochActiveAndToKeep(args)
 	var persisters []*persisterData
 	persistersMapByEpoch := make(map[uint32]*persisterData)
 
-	for epoch := int64(args.StartingEpoch); epoch >= oldestEpochKeep; epoch-- {
+	for epoch := int64(args.EpochsData.StartingEpoch); epoch >= 0; epoch-- {
+		if args.PersistersTracker.HasInitializedEnoughPersisters(epoch) {
```

Comment:
no nil check for PersistersTracker. Code can panic.

Reply:
Added on checkArgs()